packages/api/src/external/ehr/athenahealth/shared.ts

```diff
@@ -59,6 +69,43 @@ export function createNames(patient: PatientResource): { firstName: string; last
   return names;
 }
 
+export async function createAthenaClient({
+  cxId,
+  practiceId,
+  threeLeggedAuthToken,
+}: {
+  cxId: string;
+  practiceId: string;
+  threeLeggedAuthToken?: string;
+}): Promise<AthenaHealthApi> {
+  const { environment, clientKey, clientSecret } = await getAthenaEnv();
+  const twoLeggedAuthToken = await getLatestAthenaClientJwtToken({ cxId, practiceId });
+  const athenaApi = await AthenaHealthApi.create({
+    twoLeggedAuthToken,
+    threeLeggedAuthToken,
+    practiceId,
+    environment,
+    clientKey,
+    clientSecret,
+  });
+  if (!twoLeggedAuthToken) {
+    const newAuthInfo = athenaApi.getTwoLeggedAuthTokenInfo();
+    if (!newAuthInfo) throw new MetriportError("Client not created with two-legged auth token");
```

Comment:
The logic here is a bit odd:
- get twoLeggedAuthToken
- pass it to the athena API
- if it does't exist
- get it from the athena API (!)
- error if it still doesn't exist

And below, we return the athenaApi without updating it after `findOrCreateJwtToken`, is that expected?

Reply:
Updated per our discussion.